src/w3d/renderer/ddsfile.cpp

```diff
@@ -636,3 +642,47 @@ unsigned DDSFileClass::Calculate_S3TC_Surface_Size(unsigned width, unsigned heig
 
     return ret;
 }
+
+namespace Thyme
+{
+void Change_Texture_File_Extension(char *str, size_t str_capacity, const char *new_ext)
+{
+    const size_t len = strlen(str);
```

Comment:
This is just as unsafe as the previous code. If len is missing is null terminator then function writes to invalid memory.

Reply:
Do we have a strlen function with max len somewhere? I think we don't. Sounds like a separate fixing pass, as everything else uses strlen as well right now. We expect zero terminated string here. Your "unsafe" remark for the scope of this review is void.